atom/browser/native_window.cc

```diff
@@ -540,6 +573,73 @@ void NativeWindow::DevToolsClosed() {
   FOR_EACH_OBSERVER(NativeWindowObserver, observers_, OnDevToolsClosed());
 }
 
+void NativeWindow::SendKeyboardEvent(blink::WebInputEvent::Type type, int modifiers, int keycode, int nativeKeycode){
+  auto keyb_event = new content::NativeWebKeyboardEvent;
+
+  keyb_event->nativeKeyCode = nativeKeycode;
+  keyb_event->windowsKeyCode = keycode;
+  keyb_event->setKeyIdentifierFromWindowsKeyCode();
+  keyb_event->type = type;
+  keyb_event->modifiers = modifiers;
+  keyb_event->isSystemKey = false;
+  keyb_event->timeStampSeconds = base::Time::Now().ToDoubleT();
+  keyb_event->skip_in_browser = false;
+
+  if (type == blink::WebInputEvent::Char || type == blink::WebInputEvent::RawKeyDown) {
+    keyb_event->text[0] = keycode;
+    keyb_event->unmodifiedText[0] = keycode;
+  }
+
+  const auto view = web_contents()->GetRenderWidgetHostView();
+  const auto host = view ? view->GetRenderWidgetHost() : nullptr;
+  host->ForwardKeyboardEvent(*keyb_event);
+
+  if(keyb_event->type == blink::WebInputEvent::Type::KeyDown){
```

Comment:
Fixed this in an other commit, I was trying to figure out how I can "forge" valid events that produce the characters I wanted, and for some reason I didn't remove the duplicate sending - I checked, it works with a simple `keydown` event.